### PR TITLE
Improve feedback module

### PR DESCRIPTION
**Pull request by shivam-kapila**
# Problem

The module did not serve the purpose properly for the APIs.

- We had to make separate access for getting `user_name`
- We were fetching all the records and then applying offset and count at the API level.
- There was a lot of repetitive code.

# Solution

This PR aims to address these issues by:

- Using SQL JOINs to fetch `username` along with feedback records.
- Apply LIMIT and OFFSET in SQL queries.
- Remove repetitive code.

### CODE REVIEW

File: listenbrainz/db/feedback.py [hunk 1]
```diff
@@ -45,66 +47,111 @@ def delete(feedback: Feedback):
         )
 
 
-def get_feedback_by_user_id(user_id: int):
-    """ Get a list of recording feedbacks given by the user.
+def get_feedback_for_user(user_id: int, limit: int, offset: int, score: int = None):
+    """ Get a list of recording feedbacks given by the user
```
Comment (alastair) [inline listenbrainz/db/feedback.py:51]: You should add something here to explicitly say that they're returned in newest-to-oldest order.

File: listenbrainz/db/feedback.py [hunk 2]
```diff
@@ -45,66 +47,111 @@ def delete(feedback: Feedback):
...
-        )
+        result = connection.execute(sqlalchemy.text(query), args)
         return [Feedback(**dict(row)) for row in result.fetchall()]
 
 
-def get_feedback_by_user_id_and_score(user_id: int, score: int):
-    """ Get a list of recording feedbacks of particular type (loved or hated) given by the user.
-        If score is +1 then get loved recordings, if -1 then get hated recordings.
+def get_feedback_count_for_user(user_id: int):
+    """ Get total number recording feedbacks given by the user
```
Comment (alastair) [inline listenbrainz/db/feedback.py:84]: spelling: total number of recording feedbacks

File: listenbrainz/db/feedback.py [hunk 3]
```diff
@@ -45,66 +47,111 @@ def delete(feedback: Feedback):
...
+
+
+def get_feedback_count_for_recording(recording_msid: str):
+    """ Get total number recording feedbacks for a given recording
+
+        Args:
+            recording_msid: the MessyBrainz ID of the recording
+
+        Returns:
+            The total number recording feedbacks given by the user
```
Comment (alastair) [inline listenbrainz/db/feedback.py:145]: this is wrong, should be recordings?

File: listenbrainz/db/feedback.py (hunk 1 above)
Comment (shivam-kapila) [reply listenbrainz/db/feedback.py]: Added.

File: listenbrainz/db/feedback.py (hunk 2 above)
Comment (shivam-kapila) [reply listenbrainz/db/feedback.py]: Oops

File: listenbrainz/db/feedback.py (hunk 3 above)
Comment (shivam-kapila) [reply listenbrainz/db/feedback.py]: Yeah. Sorry. Corrected.

**Pull request merged**